main.py

```diff
@@ -15,10 +15,128 @@
+                   "image": "neo4j:4.2-enterprise",
+                   "version": "4.2",
+                   "edition": "enterprise",
+                   "cluster": True,
+                   "suite": "",  # TODO: Define cluster suite
+                   "scheme": "neo4j"},
+                  {"name": "3.5-enterprise",
+                   "image": "neo4j:3.5-enterprise",
+                   "version": "3.5",
+                   "edition": "enterprise",
+                   "cluster": False,
+                   "suite": "3.5",
+                   "scheme": "bolt"},
+                  {"name": "4.0-community",
+                   "image": "neo4j:4.0",
+                   "version": "4.0",
+                   "edition": "community",
+                   "cluster": False,
+                   "suite": "4.0",
+                   "scheme": "neo4j"},
+                  {"name": "4.1-enterprise",
+                   "image": "neo4j:4.1-enterprise",
+                   "version": "4.1",
+                   "edition": "enterprise",
+                   "cluster": False,
+                   "suite": "4.1",
+                   "scheme": "neo4j"},
+                  {"name": "4.2-tc-enterprise",
+                   "image": "neo4j:4.2.2-enterprise",
+                   "version": "4.2",
+                   "edition": "enterprise",
+                   "cluster": False,
+                   "suite": "4.2",
+                   "scheme": "neo4j",
+                   "download": teamcity.DockerImage("neo4j-enterprise-4.2.2-docker-loadable.tar")},
+                  {"name": "4.3-tc-enterprise",
+                   "image": "neo4j:4.3.0-drop02.0-enterprise",
+                   "version": "4.3",
+                   "edition": "enterprise",
+                   "cluster": False,
+                   "suite": "4.3",
+                   "scheme": "neo4j",
+                   "download": teamcity.DockerImage("neo4j-enterprise-4.3.0-drop02.0-docker-loadable.tar")}]
+
+
+def set_test_flags(requested_list):
+    source = []
+    if not requested_list:
+        requested_list = TestFlags
+
+    for item in TestFlags:
+        if item not in requested_list:
+            TestFlags[item] = False
+
+    print("Tests that will be run:")
+    for item in TestFlags:
+        if TestFlags[item]:
+            print("     ", item)
+
+
+def convert_to_str(input_seq, separator):
+    # Join all the strings in list
+    final_str = separator.join(input_seq)
+    return final_str
+
+
+def construct_configuration_list(requested_list):
+    # if no configs were requested we will default to adding them all
+    if not requested_list:
+        requested_list = []
+        for config in Configurations:
+            requested_list.append(config["name"])
+
+    # Now try to find the requested configs and check they are available with current teamcity status
+    for config in Configurations:
+        if config["name"] in requested_list:
+            if "download" in config:
```

Comment:
No need for this is if you check in_teamcity when you build the list, then the command line help will be right as well

Reply:
This is the building of the list. The resulting configurations_to_run list is what is used later on.

Reply:
Yes but if you build the initial list based on teamcity then invoking command line help will only list the available configs in your environment and this code can be removed. Win win